openmc/mixin.py

```diff
@@ -0,0 +1,24 @@
+import numpy as np
+
+
+class Equality(object):
+    """A Class which provides generic __eq__ and __ne__ functionality which
+    can easily be inherited by downstream classes.
+    """
+
+    def __eq__(self, other):
+        eqval = True
+        if isinstance(other, type(self)):
+            for key, value in self.__dict__.items():
+                if key in other.__dict__:
```

Comment:
The `dict.get()` method already handles cases where the key is not found (in which case it returns `None`). I think you could get rid of this `if`